ESP32LowLevel/CAN.cpp

```diff
@@ -0,0 +1,196 @@
+  int headRoll = 1;
+  canMsg.id = CAN_ID | (END_EFFECTOR_HEAD_ROLL_FEEDBACK_ID << 16);     
+  canMsg.length = 4;  
+  canMsg.rtr = 0;    
+  canMsg.extended = 1; 
+  memcpy(canMsg.data.uint8, &headRoll, 4);
+  CAN0.sendFrame(canMsg);
+
+}
+
+bool CAN::receiveMotorSetpoint() {
+    // take the first four bytes from the array and and put them in a float
+        float leftSpeed, rightSpeed;
+        memcpy(&leftSpeed, canMsg.data.uint8, 4); // Copia i primi 4 byte per leftSpeed
+        memcpy(&rightSpeed, canMsg.data.uint8 + 4, 4);
+      // in the end, uncomment these two lines
+        //motorTrLeft.setSpeed(leftSpeed);
+        //motorTrRight.setSpeed(rightSpeed);
+        }
+  
+
+
+
+
+bool CAN::receiveJointSetpoint() {
+  float yawA;
+  memcpy(&yawA, canMsg.data.uint8, 4);
+  // to be completed
+  }
+      
+
+
+// receive end effector setpoint
+bool CAN::receiveEndEffectorSetpoint(byte type) {
+   int16_t data;
+
+      switch (type) {
+        case END_EFFECTOR_PITCH_SETPOINT_ID:
+        // take the first four bytes from the array and and put them in a float
+        memcpy(&data, canMsg.data.uint8, 4);
+      // in the end, uncomment this line
+        //motorEEPitch.moveSpeed(data, SERVO_SPEED);        
+        break;
+         break;
+        case END_EFFECTOR_HEAD_PITCH_SETPOINT_ID:
+        // take the first four bytes from the array and and put them in a float
+        memcpy(&data, canMsg.data.uint8, 4);
+        // in the end, uncomment this line
+        //motorEEPitch.moveSpeed(data, SERVO_SPEED);
+                
+        break;
+        case END_EFFECTOR_HEAD_ROLL_SETPOINT_ID:
+        // take the first four bytes from the array and and put them in a float
+        memcpy(&data, canMsg.data.uint8, 4);
+        // in the end, uncomment this line
+        //motorEEPitch.moveSpeed(data, SERVO_SPEED);
+                
+        break;  
+}
+}
+
+bool CAN::receiveMessage() {
+  unsigned long time_cur = millis(); // current time
+  if (CAN0.read(canMsg)) {
+    time_data = time_cur;
+    byte type = canMsg.id >> 16;
+
+
+    // DEBUG: print frame
+      Serial.println("I'm receiving...");
+          Serial.print(canMsg.id, HEX);
+          if (canMsg.extended) {
+            Serial.print(" X ");}
+          else {
+            Serial.print(" S ");  } 
+            Serial.print(canMsg.length, DEC);
+            for (int i = 0; i < canMsg.length; i++) {
+            Serial.print(canMsg.data.byte[i], HEX);
+            Serial.print(" ");}
+            Serial.println();
+
+    
+    if (type == MOTOR_SETPOINT_ID) {
+      receiveMotorSetpoint();
+    } 
+    else if (type == END_EFFECTOR_PITCH_SETPOINT_ID || 
+             type == END_EFFECTOR_HEAD_PITCH_SETPOINT_ID || 
+             type == END_EFFECTOR_HEAD_ROLL_SETPOINT_ID) {
+      receiveEndEffectorSetpoint(type);
+    } 
+    else if (type == JOINT_YAW_SETPOINT_ID) {
+      receiveJointSetpoint();
+    }
+
+    return true;
+  }

```

Comment:
what about a `switch-case`?
```suggestion
    switch (type) {
      case MOTOR_SETPOINT_ID:
        receiveMotorSetpoint();
        break;
      case END_EFFECTOR_PITCH_SETPOINT_ID:
      case END_EFFECTOR_HEAD_PITCH_SETPOINT_ID:
      case END_EFFECTOR_HEAD_ROLL_SETPOINT_ID:
        receiveEndEffectorSetpoint(type);
        break;
      case JOINT_YAW_SETPOINT_ID:
        receiveJointSetpoint();
        break;
    }

    return true;
  }

```